models/unite-legale.ts

```diff
@@ -206,13 +206,17 @@ const getUniteLegale = async (
         // no pagination as this function is called when sirene etalab already failed
         return await fetchUniteLegaleFromInseeFallback(siren, page);
       } catch (lastFallback: any) {
+        if (lastFallback instanceof HttpNotFound) {
+          throw new SirenNotFoundError(siren);
+        }
+
         logSecondSireneInseefailed({
           siren,
           details: lastFallback.message || lastFallback,
         });
 
-        // Siren was not found in both API, return a 404
-        throw new SirenNotFoundError(siren);
+        // Siren was not found in both API, return a 500
```

Comment:
if it is not found the right code is 404. You should update the comment accordingly or use a 404 code. But this is hard to understand as it is.

Reply:
En effet le commentaire est pas bon ici je vais corriger